### PR TITLE
Starred Mob Boxes

### PR DESCRIPTION
**Pull request by AzureAaron**
- Fixes the issue with the boxes occasionally showing in some edge cases when they shouldn't by changing it to use boxes
- Bats are still given glowing cause they don't really matter that much (they always spawn in the same spots and we already have waypoints to them)

### CODE REVIEW

**Comment by kevinthegreat1** [comment]
Wait, can we use the new check, but still render the old outlines?

**Comment by Fluboxer** [comment]
> Wait, can we use the new check, but still render the old outlines?

probably, but thing to consider - after this update teammates glow according to rank which makes it slightly more confusing

Also livid glow is cancer when teammates ALSO glow

**Comment by kevinthegreat1** [comment]
The teammates part is Hypixel, not us.

**Comment by Fluboxer** [comment]
> The teammates part is Hypixel, not us.

ik, but it doesn't change fact that it looks bad now

**Comment by AzureAaron** [comment]
> Wait, can we use the new check, but still render the old outlines?

No this check is just after the game determines if the entity is eligible to be rendered or not which can also pass if the entity is behind a wall. The problem is that the entity outlines are rendered to a separate frame buffer which doesn't share the same depth values as the client's main one which is how glowing goes through walls (I think)?

I can add back the old thing as a debug option via a JVM argument and update the occlusion culling library but this is the best way to have it be accurate for now.

**Comment by kevinthegreat1** [comment]
Just looked at the pr. I don't see a point for it being a jvm arg. We can just leave the old option and maybe add a warning.

**Comment by AzureAaron** [comment]
> Just looked at the pr. I don't see a point for it being a jvm arg. We can just leave the old option and maybe add a warning.

Added the old option back, although bats will glow if either bounding boxes or glow is enabled for reasons stated above.

**Comment by AzureAaron** [comment]
For merging I'd recommend squashing it.

File: src/main/java/de/hysky/skyblocker/mixin/WorldRendererMixin.java [hunk 1]
```diff
@@ -1,20 +1,35 @@
...
 
+import de.hysky.skyblocker.skyblock.entity.MobBoundingBoxes;
 import de.hysky.skyblocker.skyblock.entity.MobGlow;
 import net.minecraft.client.render.WorldRenderer;
 import net.minecraft.entity.Entity;
 
 @Mixin(WorldRenderer.class)
 public class WorldRendererMixin {
+
+	@Inject(method = "render", slice = @Slice(from = @At(value = "INVOKE", target = "Lnet/minecraft/client/world/ClientWorld;getEntities()Ljava/lang/Iterable;"), to = @At(value = "FIELD", target = "Lnet/minecraft/entity/Entity;age:I", opcode = Opcodes.GETFIELD)), at = @At(value = "FIELD", target = "Lnet/minecraft/client/render/WorldRenderer;regularEntityCount:I", opcode = Opcodes.PUTFIELD))
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/mixin/WorldRendererMixin.java:24]: Any reason for the specific injection point. I'd just inject at `renderEntity`.

File: src/main/java/de/hysky/skyblocker/skyblock/entity/MobGlow.java [hunk 2]
```diff
@@ -44,14 +45,14 @@ public static boolean shouldMobGlow(Entity entity) {
 
 					// Regular Mobs
 					if (!(entity instanceof ArmorStandEntity)) {
-						List<ArmorStandEntity> armorStands = getArmorStands(entity);
+						List<ArmorStandEntity> armorStands = MobGlow.getArmorStands(entity);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/entity/MobGlow.java:48]:
```suggestion
						List<ArmorStandEntity> armorStands = getArmorStands(entity);
```

File: src/main/resources/assets/skyblocker/lang/en_us.json [hunk 3]
```diff
@@ -289,7 +289,9 @@
   "text.autoconfig.skyblocker.option.locations.dungeons.playerSecretsTracker": "Player Secrets Tracker",
   "text.autoconfig.skyblocker.option.locations.dungeons.playerSecretsTracker.@Tooltip": "Tracks the amount of secrets people in your dungeon run are doing.",
   "text.autoconfig.skyblocker.option.locations.dungeons.starredMobGlow": "Starred Mob Glow",
-  "text.autoconfig.skyblocker.option.locations.dungeons.starredMobGlow.@Tooltip": "Applies the glowing effect to starred mobs that are visible.",
+  "text.autoconfig.skyblocker.option.locations.dungeons.starredMobGlow.@Tooltip": "Applies the glowing effect to starred mobs that are visible.\n\n§lWARNING: This feature is experimental and you may encounter issues by using it.",
```
Comment (kevinthegreat1) [inline src/main/resources/assets/skyblocker/lang/en_us.json:292]:
```suggestion
  "text.autoconfig.skyblocker.option.locations.dungeons.starredMobGlow.@Tooltip": "Applies the glowing effect to starred mobs that are visible.\n\nWARNING: This feature is experimental and you may encounter issues with it.",
```

**Review by kevinthegreat1 — DISMISSED** [review]

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**